ApprovalTests/integrations/ut/UTApprovals.h

```diff
@@ -0,0 +1,80 @@
+#ifndef APPROVALTESTS_CPP_UTAPPROVALS_H
+#define APPROVALTESTS_CPP_UTAPPROVALS_H
+
+#include "ApprovalTests/namers/ApprovalTestNamer.h"
+
+// <SingleHpp unalterable>
+#ifdef APPROVALS_UT
+
+#include <ut.hpp>
+
+namespace ApprovalTests
+{
+    namespace cfg {
+        class reporter : public boost::ut::reporter<boost::ut::printer> {
+            private:
+                TestName currentTest;
+
+            public:
+                auto on(boost::ut::events::test_begin test_begin) -> void {
+                    std::string name = test_begin.name;
+                    currentTest.sections.emplace_back(name);
+
+                    const std::string fileName = 
+                        boost::ut::reflection::source_location::current().file_name();
+                    currentTest.setFileName(fileName);
+
+                    ApprovalTestNamer::currentTest(&currentTest);
+
+                    boost::ut::reporter<boost::ut::printer>::on(test_begin);
+                }
+
+                auto on(boost::ut::events::test_run test_run) -> void { 
```

Comment:
I wonder whether these one-liner methods that call the base class are required - it looks like it's calling the method of the same name in the base class, so could these be removed, for simplification?

No, I'm wrong - I commented out some of these in the project we experimented with yesterday, and got compiler errors - so they are required. :-)